### PR TITLE
[cdc] kafka cdc uses hive catalog to support field conversion to lowercase

### LINKED ISSUE #2014: [Feature] cdc uses hive catalog to support field conversion to lowercase
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

cdc uses hive catalog to support field conversion to lowercase

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Linked issue: close #2014

### CODE REVIEW

**Comment by yuzelin** [comment]
Can you add a test for this?

**Comment by zhuangchong** [comment]
> Can you add a test for this?

Can I add hive-related dependencies under the paimon-flink-cdc module?

**Comment by JingsongLi** [comment]
> > Can you add a test for this?
> 
> Can I add hive-related dependencies under the paimon-flink-cdc module?

No, See `TestCaseInsensitiveCatalogFactory`, it should be same to `HiveCatalog`.

**Comment by zhuangchong** [comment]
@JingsongLi @yuzelin Thanks. 

I added two new test classes:
KafkaCanalSyncDatabaseActionITCase#testCaseInsensitive
KafkaOggSyncDatabaseActionITCase#testCaseInsensitive

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaCanalSyncDatabaseActionITCase.java [hunk 1]
```diff
@@ -571,4 +576,47 @@ public void testCatalogAndTableConfig() {
...
+        kafkaConfig.put("topic", topic);
+
+        KafkaSyncDatabaseAction action =
+                syncDatabaseActionBuilder(kafkaConfig)
+                        .withTableConfig(getBasicTableConfig())
+                        .build();
+        Field catalogField = ActionBase.class.getDeclaredField("catalog");
+        catalogField.setAccessible(true);
+        Object newCatalog = catalog;
+        catalogField.set(action, newCatalog);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaCanalSyncDatabaseActionITCase.java:605]: use catalog config to set `CatalogOptions.METASTORE.key() = test-case-insensitive`

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaOggSyncDatabaseActionITCase.java [hunk 2]
```diff
@@ -462,4 +467,51 @@ private void includingAndExcludingTablesImpl(
...
+        kafkaConfig.put("topic", topic);
+
+        KafkaSyncDatabaseAction action =
+                syncDatabaseActionBuilder(kafkaConfig)
+                        .withTableConfig(getBasicTableConfig())
+                        .build();
+        Field catalogField = ActionBase.class.getDeclaredField("catalog");
+        catalogField.setAccessible(true);
+        Object newCatalog = catalog;
+        catalogField.set(action, newCatalog);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaOggSyncDatabaseActionITCase.java:496]: Use catalog-conf, too

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaCanalSyncDatabaseActionITCase.java (hunk 1 above)
Comment (zhuangchong) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/action/cdc/kafka/KafkaCanalSyncDatabaseActionITCase.java]: Thanks, I have modified it.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**